usr/share/rear/lib/_input-output-functions.sh

```diff
@@ -1177,6 +1177,28 @@ function UserInput () {
     # In debug mode show the user the script that called UserInput and what user_input_ID was specified
     # so that the user can prepare an automated response for that UserInput call (without digging in the code):
     DebugPrint "UserInput -I $user_input_ID needed in $caller_source"
+    # Drain stdin if stdin is a terminal i.e. when 'rear' is run in interactive mode
+    # where stdin is all what the user types on his keyboard.
+    # In this case discard possibly already existing characters (in particular ENTER keystrokes) from stdin
+    # to avoid that when the user had accidentally hit ENTER more than once in a previous (UserInput) dialog
+    # then those additional ENTER characters would be already in stdin and let this current UserInput dialog
+    # proceed unintendedly automatically without an explicit ENTER from the user for this current dialog,
+    # see https://github.com/rear/rear/issues/2866
+    # There is no generic way to clear stdin so we 'read' and discard what is already there
+    # up to 1000 characters (so it fails if there are more than 1000 characters in stdin)
+    # see https://superuser.com/questions/276531/clear-stdin-before-reading
+    # That the 'read' timeout can be a fractional number requires bash 4.x
+    # but in general ReaR should still work with bash 3.x so we use '-t 1'
+    # which causes a one second delay (in interactive mode) which cannot be avoided,
+    # see https://github.com/rear/rear/issues/2866#issuecomment-1254908270
+    local discard_stdin=""
+    if tty -s ; then
```

Comment:
as an alternative, there is a `-t FD` test that returns true if file descriptor FD is open and refers to a terminal. (Not sure about bash 3 though.)

Reply:
I used `tty -s` because it is already used in ReaR:
```
usr/share/rear/build/default/990_verify_rootfs.sh:
    if tty -s ; then

usr/share/rear/skel/default/etc/profile:
tty -s && test -s /etc/motd && cat /etc/motd

usr/share/rear/lib/progresssubsystem.nosh:
if tty -s <&7 ; then
```
while I cannot find usage of `test -t` or `[ -t` in ReaR
so it seems `tty -s` is the preferred "known to work way".

But `test -t` seems to also work for me
even on old SLES10-SP4 with bash version 3.1.17:
```
# test -t 0 && echo y
y
# test -t 1 && echo y
y
# test -t 2 && echo y
y

# ( test -t 0 && echo y )
y
# ( test -t 1 && echo y )
y
# ( test -t 2 && echo y )
y

# echo foo | ( test -t 0 && echo y || echo n )
n
# echo foo | ( test -t 1 && echo y || echo n )
y
# echo foo | ( test -t 2 && echo y || echo n )
y

# echo foo | ( test -t 0 && echo y || echo n ) | cat -
n
# echo foo | ( test -t 1 && echo y || echo n ) | cat -
n
# echo foo | ( test -t 2 && echo y || echo n ) | cat -
y

# { echo foo | ( test -t 2 && echo y || echo n ) | cat - ; } 2>/dev/null
n
```